### ISSUE TITLE
OpenTK is getting shutdown (see repo for more info). Should OpenGL rendering be pursued?

### ISSUE THREAD

**Issue by matthewd673** [neon2d/neon2d#52]
So, the OpenTK project is dying, with their maintainer stepping down and no way to contact the owner. Should OpenGL rendering be pursued further or is sticking with WinForms or maybe even DirectX better at this point?

**Comment by DontBelieveMe** [comment]
There are other opengl bindings for c#, although I haven't used any of them. I have also very little experience with directx, so opengl may still be the way to go. It is also cross platform compatible, which directx is not.

**Comment by matthewd673** [comment]
That is true. The only other binding I've heard of is the Tao Framework, which OpenTK expanded on. If anything Tao would be a step backwards

**Comment by DontBelieveMe** [comment]
Currently I am experimenting with creating a custom low level rendering API that is bound to C#. You can see this on 
DontBelieveMe/neon2d/E-CPP-Backend
This means all the low level rendering code can be written in low level, native, C++ code and then bound to C# for use in neon2d. This may potentially be better as it gives us more control. Also, because OpenTK shut down, there are no good/maintained C# OpenGL bindings, so creating our own may be beneficial. 
The only problem may be the speed of pinvoke (used for binding), but this can be profiled at a later date to make sure. Another problem is that this is a large undertaking. Do you have experience in C++?

**Comment by matthewd673** [comment]
Not really, I've tried it out but I'm not that great with it. Isn't it possible to use an OpenGL .dll in C# directly (something like `[DllImport]`?)

**Comment by DontBelieveMe** [comment]
Theoretically yes, but that would mean binding the whole Opengl spec, which is huge.  It would be much easier to code the OpenGL where it is easily avaliable, say a native language, and then bind those functions. It would also be considerably slower as the amount of switches between managed (C#) and unmanaged (C++) code would be much higher.
